apps/mobile/src/libs/components/radio-group/radio-group.tsx

```diff
@@ -0,0 +1,102 @@
+import React, { useState } from "react";
+import { RadioGroup as RNRadioGroup } from "react-native-radio-buttons-group";
+
+import { Text } from "~/libs/components/components";
+import { BaseColor } from "~/libs/enums/enums";
+import { globalStyles } from "~/libs/styles/styles";
+import { type RadioButtonProps, type ViewStyle } from "~/libs/types/types";
+
+import { DefaultStyles } from "./libs/constants/constants";
+import { styles } from "./styles";
+
+type RadioGroupItemStyle = {
+	circleBorderColorActive?: string;
+	circleBorderSize?: number;
+	circleBorderSizeActive?: number;
+	circleColorActive?: string;
+	circleSize?: number;
+} & ViewStyle;
+
+type RadioGroupItem = {
+	label: string;
+	value: string;
+};
+
+type Properties = {
+	itemContainerStyle?: RadioGroupItemStyle;
+	onChange?: (value: RadioGroupItem) => void;
+	options: RadioGroupItem[];
+};
+
+const RadioGroup: React.FC<Properties> = ({
+	itemContainerStyle = {},
+	onChange,
+	options,
+}: Properties) => {
+	const {
+		circleBorderColorActive = BaseColor.BG_BLUE,
+		circleBorderSize = DefaultStyles.DEFAULT_BORDER_SIZE,
+		circleBorderSizeActive = DefaultStyles.DEFAULT_ACTIVE_BORDER_SIZE,
+		circleColorActive = BaseColor.BG_WHITE,
+		circleSize = DefaultStyles.DEFAULT_CIRCLE_SIZE,
+	} = itemContainerStyle;
+
+	const [selectedId, setSelectedId] = useState<string>("");
+
+	const radioButtonsProperties: RadioButtonProps[] = [];
+	const itemsMap = new Map<string, RadioGroupItem>();
+
+	for (const [index, option] of options.entries()) {
```

Comment:
Let's use reduce() instead of a for loop. It will be more of a "TS way" to solve this issue
```
const itemsMap = options.entries().reduce((itemsMap, [index, option]) => {
...
}, {})
```

Reply:
For now we have eslint rule that doesn't allow using `reduce`
<img width="492" alt="Screenshot 2024-08-26 at 14 52 54" src="https://github.com/user-attachments/assets/2f5f0a88-412e-4e77-a991-01d91e270e7d">